v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/options/BigtableChangeStreamsToPubSubOptions.java

```diff
@@ -0,0 +1,154 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.options;
+
+import com.google.cloud.teleport.metadata.TemplateParameter;
+import com.google.cloud.teleport.v2.bigtable.options.BigtableCommonOptions;
+import org.apache.beam.runners.dataflow.options.DataflowPipelineOptions;
+import org.apache.beam.sdk.options.Default;
+import org.apache.beam.sdk.options.Validation;
+
+/**
+ * The {@link BigtableChangeStreamsToPubSubOptions} class provides the custom execution options
+ * passed by the executor at the command-line.
+ */
+public interface BigtableChangeStreamsToPubSubOptions
+    extends DataflowPipelineOptions, BigtableCommonOptions.ReadChangeStreamOptions {
+
+  @TemplateParameter.Text(
+      order = 1,
+      description = "The output Pub/Sub topic",
+      helpText = "The Pub/Sub topic to publish changelog entry messages.")
+  @Validation.Required
+  String getPubSubTopic();
+
+  void setPubSubTopic(String pubSubTopic);
+
+  @TemplateParameter.Text(
+      order = 2,
+      optional = true,
+      description = "The encoding of the message written into PubSub",
+      helpText =
+          "The format of the message to be written into PubSub. "
+              + "Allowed formats are BINARY and JSON Text. Default value is JSON.")
+  @Default.String("JSON")
+  String getMessageEncoding();
+
+  void setMessageEncoding(String messageEncoding);
+
+  @TemplateParameter.Text(
+      order = 3,
+      optional = true,
+      description = "The format of the message written into PubSub",
+      helpText =
+          "The message format chosen for outputting data to PubSub. "
+              + "Allowed formats are AVRO, PROTOCOL_BUFFERS and JSON Text. Default value is JSON.")
```

Comment:
Looks like there is some way to make it validated by the engine, it's a new feature, I haven't used myself, but DF templates people might call this out:

https://github.com/GoogleCloudPlatform/DataflowTemplates/blob/main/v2/googlecloud-to-googlecloud/src/main/java/com/google/cloud/teleport/v2/options/DataplexBigQueryToGcsOptions.java#L102-L109